storage/badger/cluster_payloads.go

```diff
@@ -53,6 +53,9 @@ func (cp *ClusterPayloads) storeTx(blockID flow.Identifier, payload *cluster.Pay
 func (cp *ClusterPayloads) retrieveTx(blockID flow.Identifier) func(*badger.Txn) (*cluster.Payload, error) {
 	return func(tx *badger.Txn) (*cluster.Payload, error) {
 		val, err := cp.cache.Get(blockID)(tx)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This was added because when testing the method was trying to convert `val` which is `nil` if nothing was stored to `*cluster.Payload` which was causing errors.